technique/reflectometry/NR_motion.py

```
@@ -565,17 +506,22 @@ def setup_measurement(self, mode=None, periods=1):
         """
         self.dry_run_warning()
         constants = get_instrument_constants()
-        self.change_to_soft_period_count(count=periods)
-        try: 
+        ##TODO: this constants path might need changing depending how constants setup. Does it need to be set separately here?
+        if periods is not None:
+            self.change_to_soft_period_count(count=periods)
+        else:
+            print("Periods not changed as no default periods set in instrument constants.")
+        try:
             g.cset("MONITOR", "IN")
         except:
```

We should probably go through the code and modify the except statements such that they are working on a specific exception. Working on a bare except can make debugging very difficult.